### PR TITLE
chore(smp upgrade) - Upgrade SMP to latest release

### PR DESCRIPTION
**Pull request by scottopell**

### CODE REVIEW

File: .github/workflows/workload_checks.yml
Comment (GeorgeHahn) [inline .github/workflows/workload_checks.yml:1]: Could you also delete the `workload-checks` top level directory?

File: regression/cases/fluent_elasticsearch/lading/lading.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ generator:
   - tcp:
       seed: [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53,
         59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103, 107, 109, 113, 127, 131]
-      addr: "localhost:8282"
+      addr: "0.0.0.0:8282"
```
Comment (GeorgeHahn) [inline regression/cases/fluent_elasticsearch/lading/lading.yaml:5]: I'm somewhat surprised that `0.0.0.0` works here; lading is connecting to this and I didn't know Linux supported connections to that address. (I'd have expected `127.0.0.1` here.) No change needed, just curious.

File: regression/config.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# - `end_time_ms`: end time of the job's metrics data, in ms
+# - `filter_start`: arbitrary time before this job's replicates were run,
+#   suitable for filtering logs and profiles by job ID.
+# - `filter_end`: arbitrary time after this job's replicates were run and logs
+#   were uploaded, suitable for filtering logs and profiles by job ID.
+#
+# Additional variables for per-experiment links:
+# - `experiment`: the name of the experiment
+report:
+  metrics_dashboard: "https://app.datadoghq.com/dashboard/ykh-ua8-vcu/SMP-Regression-Detector-Metrics?fromUser=true&refresh_mode=paused&tpl_var_run-id%5B0%5D={{ job_id }}&view=spans&from_ts={{ start_time_ms }}&to_ts={{ end_time_ms }}&live=false"
```
Comment (GeorgeHahn) [inline regression/config.yaml:37]: Would you mind also adding a logs link?

**Review by GeorgeHahn — APPROVED** [review]

File: .github/workflows/regression.yml [hunk 3]
```diff
@@ -121,19 +121,9 @@ jobs:
...
-
-      replicas: ${{ steps.experimental-meta.outputs.REPLICAS }}
-      warmup-seconds: ${{ steps.experimental-meta.outputs.WARMUP_SECONDS }}
-      total-samples: ${{ steps.experimental-meta.outputs.TOTAL_SAMPLES }}
-      p-value: ${{ steps.experimental-meta.outputs.P_VALUE }}
       smp-version: ${{ steps.experimental-meta.outputs.SMP_CRATE_VERSION }}
-      lading-version: ${{ steps.experimental-meta.outputs.LADING_VERSION }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```
Comment (jszwedko) [inline .github/workflows/regression.yml:126]: We actually withheld this version bump in https://github.com/vectordotdev/vector/pull/18490 but I think it's ok to do for this workflow.

File: regression/cases/fluent_elasticsearch/lading/lading.yaml (hunk 1 above)
Comment (scottopell) [reply regression/cases/fluent_elasticsearch/lading/lading.yaml:5]: It is possible that this specific one wasn't needed, I just took a "big hammer" approach and made everything `0.0.0.0`

File: regression/cases/datadog_agent_remap_blackhole/experiment.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+target:
+  name: vector
+  command: /usr/bin/vector
+  cpu_allotment: 7
+  memory_allotment: 30g
+
+  environment:
+    VECTOR_THREADS: 4
+
+
```
Comment (jszwedko) [inline regression/cases/datadog_agent_remap_blackhole/experiment.yaml:12]:
```suggestion
```

Nit: extra whitespace 😄

File: regression/cases/fluent_elasticsearch/lading/lading.yaml (hunk 1 above)
Comment (jszwedko) [reply regression/cases/fluent_elasticsearch/lading/lading.yaml:5]: I'm also a bit surprised by this. Did you find this change necessary? I could see it causing confusion in the future. I'd also expect `localhost` here.

**Review by jszwedko — APPROVED** [review]
I left a few nits/questions, but overall 👍

File: regression/cases/fluent_elasticsearch/lading/lading.yaml (hunk 1 above)
Comment (scottopell) [reply regression/cases/fluent_elasticsearch/lading/lading.yaml:5]: I'll try with `localhost` in a followup PR to see.

Comment (GeorgeHahn) [reply regression/cases/fluent_elasticsearch/lading/lading.yaml:5]: The hostname `localhost` is broken inside target containers (never investigated; it's a longstanding bug that'd be really nice to have fixed.)

**Pull request merged**